hyperspy/_components/logistic.py

```diff
@@ -17,87 +17,58 @@
+class Logistic(Expression):
 
-    """Logistic function component
+    """Logistic function (sigmoid/s-shaped curve) component
 
-    f(x) = a/(1+b*exp(-c*(x-origin)))
+    .. math::
+    
+        f(x) = a / (1 + b * exp(-c * (x - origin)))
 
-    Attributes
-    ----------
+    Parameters
+    -----------
     a : Float
+        The curve's maximum y-value, lim(x->infty)=a
     b : Float
+        Additional parameter: 
+        b>1 shifts origin to larger values
+        0<b<1 shifts origin to smaller values
+        b<0 introduces an asymptote
     c : Float
+        logistic growth rate or steepness of the curve
     origin : Float
+        position of the sigmoid's midpoint
 
     """
 
-    def __init__(self):
-        # Define the parameters
-        Component.__init__(self, ('a', 'b', 'c', 'origin'))
-        # Define the name of the component
-        self.a.grad = self.grad_a
-        self.b.grad = self.grad_b
-        self.c.grad = self.grad_c
-        self.origin.grad = self.grad_origin
-        self._position = self.origin
-
-    def function(self, x):
-        """
-        """
-        a = self.a.value
-        b = self.b.value
-        c = self.c.value
-        origin = self.origin.value
-        return a / (1 + b * np.exp(-c * (x - origin)))
-
-    def grad_a(self, x):
-        """
-        Returns d(function)/d(parameter_1)
-        """
-        b = self.b.value
-        c = self.c.value
-        origin = self.origin.value
-
-        return 1 / (1 + b * np.exp(-c * (x - origin)))
-
-    def grad_b(self, x):
-        """
-        Returns d(function)/d(parameter_1)
-        """
-        a = self.a.value
-        b = self.b.value
-        c = self.c.value
-        origin = self.origin.value
-
-        return -(a * np.exp(-c * (x - origin))) / \
-            (b * np.exp(-c * (x - origin)) + 1) ** 2
-
-    def grad_c(self, x):
-        """
-        Returns d(function)/d(parameter_1)
-        """
-        a = self.a.value
-        b = self.b.value
-        c = self.c.value
-        origin = self.origin.value
-
-        return -(a * b * (origin - x) * np.exp(-c * (x - origin))) / \
-            (b * np.exp(-c * (x - origin)) + 1) ** 2
-
-    def grad_origin(self, x):
-        """
-        Returns d(function)/d(parameter_1)
-        """
-        a = self.a.value
-        b = self.b.value
-        c = self.c.value
-        origin = self.origin.value
-
-        return -(a * b * c * np.exp(-c * (x - origin))) / \
-            (b * np.exp(-c * (x - origin)) + 1) ** 2
+    def __init__(self, a=1., b=1., c=1., origin=0., module="numexpr", **kwargs):
+        super(Logistic, self).__init__(
+            expression="a / (1 + b * exp(-c * (x - origin)))",
+            name="Logistic",
+            a=a,
+            b=b,
+            c=c,
+            origin=origin,
+            position="origin",
+            module=module,
+            autodoc=False,
+            **kwargs)
+
+        # Boundaries
+        self.a.bmin = None
```

Comment:
Is there any reason to set explicitly `bmin`, `bmax` to `None` for the different parameters? `None` should be already the default?

Reply:
No, so removed.